### DISCUSSION TITLE
OpenSSL application (/usr/bin/openssl) support the engine for loading the SM2 algorithm from the command line?

### DISCUSSION THREAD

**Discussion by linux-gcc** [openssl/openssl#21670] [Q&A]
Please do NOT use issues to ask questions about OpenSSL.

Instead, please use the [Q&A category in Discussions](<https://github.com/openssl/openssl/discussions/new?category=q-a>)
to ask your question.

**Comment by paulidale** [comment]
OpenSSL 1.1.1 and 3.x have SM2 built in.
No need for an engine.

**Reply by linux-gcc** [reply]
I want to use SM2's hard algorithm via the openssl application (/usr/bin/openssl). Is it now possible to load the SM2 hard algorithm engine via the openssl command line?
